modules/seqexec/web/client/src/main/scala/seqexec/web/client/components/LogArea.scala

```diff
@@ -82,66 +109,155 @@ object LogArea {
     val reverseLog: FixedLengthBuffer[ServerLogMessage] = log().log.reverse
 
     // Filter according to the levels on the controls
-    private def levelFilter(s: State)(m: ServerLogMessage): Boolean = s.allowedLevel(m.level)
+    private def levelFilter(s: State)(m: ServerLogMessage): Boolean =
+      s.allowedLevel(m.level)
 
-    def rowGetter(s: State)(i: Int): LogRow = reverseLog.filter_(levelFilter(s) _).lift(i).map { l =>
-        val localTime = LocalDateTime.ofInstant(l.timestamp, site.timezone)
-        LogRow(formatter.format(localTime), l.timestamp, l.level, l.msg)
-      }.getOrElse(LogRow.Zero)
+    def rowGetter(s: State)(i: Int): LogRow =
+      reverseLog
+        .filter_(levelFilter(s) _)
+        .lift(i)
+        .map { l =>
+          val localTime = LocalDateTime.ofInstant(l.timestamp, site.timezone)
+          LogRow(formatter.format(localTime), l.timestamp, l.level, l.msg)
+        }
+        .getOrElse(LogRow.Zero)
 
     def rowCount(s: State): Int =
       reverseLog.filter_(levelFilter(s) _).size
 
   }
 
-  final case class State(selectedLevels: Map[ServerLogLevel, Boolean]) {
-    def updateLevel(level: ServerLogLevel, value: Boolean): State = copy(selectedLevels + (level -> value))
-    def allowedLevel(level: ServerLogLevel): Boolean = selectedLevels.getOrElse(level, false)
+  final case class State(selectedLevels: Map[ServerLogLevel, Boolean],
+                         tableState: TableState[TableColumn]) {
+
+    def updateLevel(level: ServerLogLevel, value: Boolean): State =
+      copy(selectedLevels + (level -> value))
+
+    def allowedLevel(level: ServerLogLevel): Boolean =
+      selectedLevels.getOrElse(level, false)
   }
 
   object State {
-    val Zero: State = State(ServerLogLevel.all.map(_ -> true).toMap)
+
+    val ZeroTableState: TableState[TableColumn] = TableState[TableColumn](
+      userModified   = NotModified,
+      scrollPosition = 0,
+      columns        = NonEmptyList.of(TimestampColumnMeta,
+                                       LevelColumnMeta,
+                                       MsgColumnMeta,
+                                       ClipboardColumnMeta))
+
+    val Zero: State =
```

Comment:
When I see a member called `Zero` I expect it to be a zero with respect to some operator (i.e., a monoidal zero) so if it's not a monoid maybe something like `Initial` or `Default` would express the meaning more clearly.